src/FormBuilder/Routes/CreateFormRoute.php

```diff
@@ -25,12 +27,18 @@ class CreateFormRoute
     public function __invoke()
     {
         if (isset($_GET['page']) && FormBuilderRouteBuilder::SLUG === $_GET['page']) {
+            $locale = $_GET['locale'] ?? '';
+
             // Little hack for alpha users to make sure the form builder is loaded.
             if (!isset($_GET['donationFormID'])) {
-                wp_redirect(FormBuilderRouteBuilder::makeCreateFormRoute());
+                wp_redirect(FormBuilderRouteBuilder::makeCreateFormRoute($locale)->getUrl());
                 exit();
             }
             if ('new' === $_GET['donationFormID']) {
+                // Make sure the Form will be created using the proper locale
+                $locale = $_GET['locale'] ?? '';
+                Language::switchToLocale($locale);
```

Comment:
@glaubersilva is the form actually being created for a specific locale? Or is this a convenience for translating values for form settings?

I'm wondering if we are coupling locale too closely to the forms.

Reply:
@kjohnson This Is just a convenience  for translating values as described here: https://github.com/impress-org/givewp/pull/7710#discussion_r1952884538

However, if you use the "Add Form" button from the legacy list table, the post will be created in the selected language because it's handled by these multilingual plugins. So, in the legacy list view, also is possible to filter the posts (only the ones created through the legacy view) by language using this language selector from the WP admin bar.

Reply:
In part my concern is an icrease in scope. The goal here was to add suppoprt for locale in v3 views and routes so that the forms are translatable. Modifying how a form is created was not accounted for in refinment and prioritization.